flocker/control/test/test_protocol.py

```diff
@@ -974,20 +1078,16 @@ def test_logging(self, logger):
         its connections.
         """
         control_amp_service = build_control_amp_service(self)
-        connection_protocol = ControlAMP(Clock(), control_amp_service)
-        connection_protocol.makeConnection(StringTransport())
 
-        # Patching is bad.
-        # https://clusterhq.atlassian.net/browse/FLOC-1603
-        self.patch(
-            connection_protocol,
-            "callRemote",
-            lambda *args, **kwargs: succeed({})
-        )
+        agent = FakeAgent()
+        client = AgentAMP(Clock(), agent)
+        server = LoopbackAMPClient(client.locator)
+
+        control_amp_service.connected(server)
+
         self.patch(control_amp_service, 'logger', logger)
 
-        control_amp_service._send_state_to_connections(
-            connections=[connection_protocol])
+        control_amp_service._send_state_to_connections(connections=[server])
 
         assertHasAction(
```

Comment:
This test has failed on a couple of builds on this assertion with no action logged (e.g. http://build.clusterhq.com/builders/flocker%2Funit-test%2Fcentos-7%2Fzfs-head/builds/1467/steps/trial/logs/stdio) but I can't reproduce locally or see any immediately obvious cause.